internal/lib/labels/labels.go

```diff
@@ -0,0 +1,84 @@
+package labels
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+)
+
+const (
+	maxLabelsLength   = 32
+	maxLabelsKVLength = 63
+)
+
+const (
+	keyType   = "key"
+	valueType = "value"
+)
+
+var errNilLabels = errors.New("labels is nil")
+
+var (
+	kvInvalidChar       = regexp.MustCompile(`[^a-z0-9\-_]+`)
+	kvValidStartEndChar = regexp.MustCompile(`^[a-z0-9].*[a-z0-9]$`)
+)
+
+type Labels map[string]string
+
+func (l Labels) Validate() error {
+	if l == nil {
+		return errNilLabels
+	}
+
+	if len(l) > maxLabelsLength {
+		return fmt.Errorf("labels length is more than [%d]", maxLabelsLength)
+	}
+
+	for key, value := range l {
+		if err := l.validateKV(key, keyType); err != nil {
```

Comment:
instead of passing `_type` arguments, could we separate into 2 different functions?

Reply:
previous implementations, it was actually separated. but, upon checking, turned a duplication of implementation since both key and value have the same requirements. considerations for this, either we allow for duplication (which can doubtfully introduce inconsistencies) or we extract to a function. though, I kinda agree we might need to update the `_type` part maybe to a constant type or something.

Reply:
updated as internally discussed